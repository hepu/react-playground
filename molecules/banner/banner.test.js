import React from "react"
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer'
import Banner from "./banner"
import { configure } from 'enzyme';
import 'jest-styled-components'

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('<Banner />', () => {
  it('renders 2 <p> tags', () => {
    // const wrapper = shallow(<Banner />);
    // expect(wrapper.find("p").length).toBe(2);
    const renderedWrapper = mount(<Banner />)
    expect(renderedWrapper.find("p").length).toBe(2);
    expect(renderedWrapper).toHaveStyleRule('background-color', 'red');
  });

  it('ok with renderer', () => {
    const tree = renderer.create(<Banner />).toJSON()
    expect(tree).toMatchSnapshot()
  })
});
