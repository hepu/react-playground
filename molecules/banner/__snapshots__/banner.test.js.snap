// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> ok with renderer 1`] = `
.c0 {
  background-color: red;
  height: 200px;
  width: 200px;
}

.c1 {
  font-size: 16px;
}

<div
  className="c0"
>
  <p
    className="c1"
  >
    This is a paragraph
  </p>
  <p
    className="c1"
  >
    This is another paragraph
  </p>
</div>
`;
